android/src/main/java/com/swmansion/rnscreens/bottomsheet/DimmingFragment.kt

```diff
@@ -0,0 +1,450 @@
+
+        val rootView = screen.reactContext!!.currentActivity!!.window.decorView
+        ViewCompat.setOnApplyWindowInsetsListener(rootView, this)
+//        ViewCompat.setWindowInsetsAnimationCallback(rootView, insetCallback)
+//        ViewCompat.setWindowInsetsAnimationCallback(containerView, insetCallback)
+    }
+
+    private class AnimateDimmingViewCallback(val screen: Screen, val viewToAnimate: View, val maxAlpha: Float) : BottomSheetCallback() {
+        private var largestUndimmedOffset: Float = computeOffsetFromDetentIndex(screen.sheetLargestUndimmedDetentIndex)
+        private var firstDimmedOffset: Float = computeOffsetFromDetentIndex((screen.sheetLargestUndimmedDetentIndex + 1).coerceIn(0, screen.sheetDetents.count() - 1))
+        private var intervalLength = firstDimmedOffset - largestUndimmedOffset
+        private var animator = ValueAnimator.ofFloat(0F, maxAlpha).apply {
+            duration = 1
+            addUpdateListener {
+                viewToAnimate.alpha = it.animatedValue as Float
+            }
+        }
+
+        override fun onStateChanged(bottomSheet: View, newState: Int) {
+            if (newState == BottomSheetBehavior.STATE_DRAGGING || newState == BottomSheetBehavior.STATE_SETTLING) {
+                largestUndimmedOffset = computeOffsetFromDetentIndex(screen.sheetLargestUndimmedDetentIndex)
+                firstDimmedOffset = computeOffsetFromDetentIndex((screen.sheetLargestUndimmedDetentIndex + 1).coerceIn(0, screen.sheetDetents.count() - 1))
+                assert(firstDimmedOffset >= largestUndimmedOffset) { "fdo: $firstDimmedOffset, luo: $largestUndimmedOffset" }
+                intervalLength = firstDimmedOffset - largestUndimmedOffset
+            } else if (newState != BottomSheetBehavior.STATE_HIDDEN) {
+            }
+        }
+
+        @RequiresApi(Build.VERSION_CODES.LOLLIPOP_MR1)
+        override fun onSlide(bottomSheet: View, slideOffset: Float) {
+
+            if (largestUndimmedOffset < slideOffset && slideOffset < firstDimmedOffset) {
+                val fraction = (slideOffset - largestUndimmedOffset) / intervalLength
+                animator!!.setCurrentFraction(fraction)
+            }
+        }
+
+        private fun computeOffsetFromDetentIndex(index: Int): Float {
+            return when (screen.sheetDetents.size) {
+                1 -> when (index) {
+                    -1 -> -1F
+                    0 -> 1F
+                    else -> -1F
+                }
+                2 -> when (index) {
+                    -1 -> -1F
+                    0 -> 0F
+                    1 -> 1F
+                    else -> -1F
+                }
+                3 -> when (index) {
+                    -1 -> -1F
+                    0 -> 0F
+                    1 -> screen.sheetBehavior!!.halfExpandedRatio
+                    2 -> 1F
+                    else -> -1F
+                }
+                else -> -1F
+            }
+        }
+    }
```

Comment:
I mean, this is simple mapping. I did not use some hash maps here, because in one case I need to have dynamic value.

I'll add a comment explaining what's going on here exactly.